3p/ampcontext.js

```diff
@@ -29,6 +29,9 @@ export class AbstractAmpContext {
     dev().assert(!this.isAbstractImplementation_(),
         'Should not construct AbstractAmpContext instances directly');
 
+    /** @visibleForTesting */
+    this._ampInternalIsNewImpl = true;
```

Comment:
If so, i suppose you set this boolean to false somewhere right?
Although I still think experiment flag should work for you.